### PR TITLE
Update package and testing configurations

### PR DESCRIPTION
**Pull request by dkirkham**
This is a continuation of #472 to address #459.

In particular, this:
- removes Wagtail 4.2 testing support
- removes the Django 4.1 classifier
- add the Wagtail 6 classifier

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -69,9 +69,9 @@
         'Programming Language :: Python :: 3.12',
         'Framework :: Django',
         'Framework :: Django :: 3.2',
-        'Framework :: Django :: 4.1',
```
[the remark this answers is not in the document]
Comment (dkirkham) [reply setup.py:72]: According to https://www.djangoproject.com/download/#supported-versions, Django 4.1 dropped out of extended support in December 2023. I think we should advise against any new usage.

It was an oversight that there are still tests against Django 4.1. I can update this PR by removing the `wt52-dj41-py{38,39,310,311}` line in `.tox`, or to put the classifier back in.

[the remark this answers is not in the document]
Comment (MrCordeiro) [reply setup.py:72]: My bad: in my mind it was still 2023 😅

In that case, yes, it makes sense to remove the tests.

[the remark this answers is not in the document]
Comment (dkirkham) [reply setup.py:72]: Unnecessary Django 4.1 tests removed.

**Review by MrCordeiro — APPROVED** [review]

**Comment by MrCordeiro** [comment]
Feel free to merge! 💪

**Review by schlich — APPROVED** [review]

**Pull request merged**